staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -121,6 +121,202 @@ const (
+	// line breaks. The message must not contain line breaks.
+	// If unset, the message is "failed rule: {Rule}".
+	// e.g. "must be a URL with the host matching spec.host"
+	// Optional.
+	Message string `json:"message" protobuf:"bytes,3,opt,name=message"`
+	// The Message retured when failed the validation.
+	// Optional.
+	ExpressMessage string `json:"expressMessage" protobuf:"bytes,4,opt,name=expressMessage"`
+	// Reason returned when failed the validation.
+	// +optional
+	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	// Code returned when failed the validation.
+	Code string `json:"code" protobuf:"bytes,6,opt,name=code"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PolicyBinding binds the ValidatingAdmissionPolicy with paramerized resources.
+// PolicyBinding and parameter CRDs together define how cluster administrators configure policies for clusters.
+type PolicyBinding struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Specification of the desired behavior of the AdmissionPolicy.
+	// +optional
+	Spec PolicyBindingSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PolicyBindingList is a list of PolicyBinding.
+type PolicyBindingList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// List of PolicyBinding.
+	Items []PolicyBinding `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PolicyBindingSpec is the specification of the PolicyBinding.
+type PolicyBindingSpec struct {
+	// Policy references a ValidatingAdmissionPolicy which the policyBinding binds to.
+	// Required.
+	Policy string `json:"policy,omitempty" protobuf:"bytes,1,rep,name=policy"`
+
+	//Param specifies the parameter resource used to configure the admission control policy.
+	// It should point to a Customer Resource which is created out of the CRD specified in ParamSource of ValidatingAdmissionPolicy it binded.
+	// +optional
+	Param string `json:"param,omitempty" protobuf:"bytes,2,rep,name=param"`
+
+	//MatchResources describe a list of match options which could be used to filter the resources.
+	// +optional
+	MatchResources MatchResources `json:"matchResources,omitempty" protobuf:"bytes,3,rep,name=matchResources"`
+}
+
+// MatchResources decides whether to run the admission control policy on an object based
+// on whether meet the match criteria.
+type MatchResources struct {
```

Comment:
We need `MatchPolicy`. It makes sense to me to put it here inside of `MatchResources`.